File: sdk/python/kazer.py
"""Minimal Kazer Universal SDK for the public MVP."""
from __future__ import annotations
import json
from urllib.request import Request, urlopen

class Kazer:
    def __init__(self, api_key: str | None = None, base_url: str = "http://localhost:3000"):
        self.api_key = api_key
        self.base_url = base_url.rstrip("/")

    def _request(self, path: str, payload: dict | None = None) -> dict:
        data = json.dumps(payload).encode() if payload is not None else None
        headers = {"Content-Type": "application/json"}
        if self.api_key:
            headers["Authorization"] = f"Bearer {self.api_key}"
        response = urlopen(Request(self.base_url + path, data=data, headers=headers, method="POST" if data else "GET"))
        return json.loads(response.read())

    def chat(self, prompt: str | None = None, messages: list[dict] | None = None, model: str | None = None) -> dict:
        body = {"messages": messages} if messages else {"prompt": prompt}
        if model: body["model"] = model
        return self._request("/v1/chat", body)

    def capabilities(self) -> dict: return self._request("/v1/capabilities")
    def registry(self) -> dict: return self._request("/v1/registry")
    def skills(self) -> dict: return self._request("/v1/skills")
    def auto(self, task: str) -> dict: return self.chat(prompt=task)

File: sdk/typescript/kazer.ts
export type KazerOptions = { apiKey?: string; baseURL?: string };
export type ChatRequest = { prompt?: string; messages?: Array<{ role: string; content: string }>; model?: string; max_tokens?: number };

export class Kazer {
  private readonly baseURL: string;
  private readonly apiKey?: string;
  constructor(options: KazerOptions = {}) { this.baseURL = (options.baseURL ?? "http://localhost:3000").replace(/\/$/, ""); this.apiKey = options.apiKey; }
  private async request<T>(path: string, init?: RequestInit): Promise<T> {
    const response = await fetch(`${this.baseURL}${path}`, { ...init, headers: { "content-type": "application/json", ...(this.apiKey ? { authorization: `Bearer ${this.apiKey}` } : {}), ...(init?.headers ?? {}) } });
    if (!response.ok) throw new Error(`Kazer API ${response.status}: ${await response.text()}`);
    return response.json() as Promise<T>;
  }
  chat(input: ChatRequest | string) { return this.request<{ id: string; output: string }>("/v1/chat", { method: "POST", body: JSON.stringify(typeof input === "string" ? { prompt: input } : input) }); }
  capabilities() { return this.request<{ data: unknown[] }>("/v1/capabilities"); }
  registry() { return this.request<{ data: unknown[] }>("/v1/registry"); }
  skills() { return this.request<{ data: unknown[] }>("/v1/skills"); }
  auto(task: string) { return this.chat({ prompt: task }); }
}
export default Kazer;
